### PR TITLE
Remove str_to_string lint

### PR DESCRIPTION
**Pull request by Manishearth**
Specialization makes all of the options equally efficient.

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/10470)

### CODE REVIEW

File: components/plugins/lib.rs [hunk 1]
```diff
@@ -70,5 +70,4 @@ fn register_clippy(reg: &mut Registry) {
 }
 #[cfg(not(feature = "clippy"))]
 fn register_clippy(reg: &mut Registry) {
```
Comment (KiChjang) [inline components/plugins/lib.rs]: Should this function be removed altogether? After your changes, it does nothing.

Comment (Manishearth) [inline components/plugins/lib.rs]: No, it's still necessary. We don't `#![plugin(clippy)]` on all our crates, we just register it here as if the clippy lints are our own.

**Comment by Ms2ger** [comment]
I kinda liked that we decided on a single method, even if they're equally fast...

**Pull request merged**